guides/developer/plugin-creators/deck-plugin.md

```diff
@@ -0,0 +1,451 @@
+
+1. Run `orca` and the `pf4jStagePlugin` using the **Build and Test Plugin**  configuration. On successful launch, you see a "Completed initialization" log statement in the console:
+
+   ```bash
+	020-05-12 16:03:44.274  INFO 6973 --- [0.0-8083-exec-1] o.a.c.c.C.[Tomcat].[localhost].[/]       : [] Initializing Spring DispatcherServlet 'dispatcherServlet'
+   INFO 6973 --- [0.0-8083-exec-1] o.s.web.servlet.DispatcherServlet        : [] Initializing Servlet 'dispatcherServlet'
+   INFO 6973 --- [0.0-8083-exec-1] o.s.web.servlet.DispatcherServlet        : [] Completed initialization in 16 ms
+	```
+
+   If you see error messages about Redis, make sure all the pods in your Spinnaker instance are `READY`. You can check the IP address and port for each service in `~/.spinnaker/spinnaker-local.yml`.
+
+	Plugin loading messages appear near the top of the Orca log. You should see statements similar to:
+
+	```bash
+   INFO 90843 --- [main] org.pf4j.AbstractPluginManager: [] Plugin 'Armory.RandomWaitPlugin@unspecified' resolved
+   INFO 90843 --- [main] org.pf4j.AbstractPluginManager: [] Start plugin 'Armory.RandomWaitPlugin@unspecified'
+   INFO 90843 --- [main] i.a.p.s.wait.random.RandomWaitPlugin: [] RandomWaitPlugin.start()
+	```
+
+## Build and run Deck
+
+The Deck project [README](https://github.com/spinnaker/deck) has instructions for building and running Deck locally.
+
+1. Build Deck by executing `yarn` from the `deck` directory.
+2. Start Deck with the API_HOST argument, which is the Gate URL.
+
+   ```bash
+   cd deck
+   yarn
+   API_HOST=http://192.168.64.5:8084 yarn start
+	```
+
+## Verify the plugin loads in Deck
+
+1. Access the the Spinnaker UI at `http://localhost:9000`.
+1. Go to **Applications** > **spin** > **PIPELINES**.
+1. Create a new pipeline.
+1. Add a new stage.
+1. Look for "Random Wait" in the **Type** select list.
+
+### Troubleshooting
+
+You can use the Developer Tools in your browser to troubleshoot Deck plugin issues.
+
+![Debugging Deck in Chromium](/assets/images/guides/developer/plugin-creators/debugDeck01.png)
+
+Look for `plugin-manifest.json` and `index.js`. It's normal to see 3 `plugin-manifest.json` files. The first one is from Deck - you created this file in the [Configure Deck for the plugin](#configure-deck-for-the-plugin) section above. The next two from Gate are not relevant for local development. If you don't see the `plugin-manifest.json` and `index.js` files, check the **Console** tab for errors. Also verify that the content in your `plugin-manifest.json` file is correct.
+
+## Debug the backend of the plugin
+
+If you want to debug the backend component of the plugin without a working Deck component, you can create a new stage and configure it using JSON.
+
+1. Start the **Build and Test Plugin** configuration in Debug mode.
+1. Start Deck.
+1. Access the the Spinnaker UI at `http://localhost:9000`.
+1. Go to **Applications** > **spin** > **PIPELINES**.
+1. Create a new pipeline.
+1. Add a new stage.
```

Comment:
Might say "Add any new stage" and maybe even give an example so that it's clear that it can be a deploy stage or whatever.